.github/workflows/ci.yml

```diff
@@ -0,0 +1,53 @@
+name: Continuous Integration
+on:
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+    branches:
+      - master
+  push:
+    tags:
+      - '**'
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build-run-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out repository contents
+        uses: actions/checkout@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+      - name: Build binary
+        run: |
+          docker build  --tag lustre_exporter:binary -f docker/Dockerfile .
+      - name: Build rpm
+        run: |
+          docker build -t lustre_exporter:rpm -f docker/RPM-Dockerfile .
+
+      - name: Extract binary
+        if: github.event_name != 'pull_request'
```

Comment:
Well my initial idea was: the pipeline should run every time someone opens a PR of to check if that builds.
Also if someone push or merges to master the pipeline should run, so that the badge gets updated.
Then if someone pushes a tag the pipeline should also upload the results and store them as artifacts so that one can download it.

So maybe in the end we should go with something like:

```
on:
  pull_request:
    types:
      - opened
      - reopened
      - synchronize
    branches:
      - master
  push:
    branches:
      - master
[...]
      - name: Extract rpm
        if: startsWith(github.ref, 'refs/tags/v')
[...]
```

Reply:
Yes, agree with you.  

I would suggest the following, so all tags get also build:  

```yaml
on:
  pull_request:
    types:
      - opened
      - reopened
      - synchronize
    branches:
      - master
  push:
    branches:
      - master
    tags:
      - '**'
```

The other part is the extract/upload of binary/RPM package.  
Also agree with you here, this should be done for tags only.  

So we could use:   

```yaml
- name: Extract rpm
        if: startsWith(github.ref, 'refs/tags/v')
```


If you could update and also test the `push on all tags` please,   
we should be ready to go :-).

Best 
Gabriele

Reply:
done, I changed the CI as requested and tested it.
The CI is run when I push to master, open a PR and when pushing a tag.

![Screenshot 2024-04-26 at 15 57 26](https://github.com/GSI-HPC/lustre_exporter/assets/1274814/c78d127e-fe5e-40c0-a35c-b2f674634241)

Reply:
great! thanks